### PR TITLE
TF: XLA Logits Warpers

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR enables XLA on the logits warpers... which actually needed no changes. In essence, it adds XLA tests to ensure we don't regress.

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]

**Comment by gante** [comment]
@patrickvonplaten Sorry, I know you've already reviewed this, but I'm going to re-request your review. I realized the tests were much easier to understand (and with fewer lines) if they were parametrized, instead of having two tests (one for XLA, another for non-XLA) with shared code 😅

File: tests/generation/test_generation_tf_logits_process.py [hunk 1]
```diff
@@ -101,7 +106,8 @@ def test_temperature_dist_warper(self):
         self.assertGreater(tf.math.reduce_max(probs[1, :]), tf.math.reduce_max(warped_prob_smooth[1, :]))
         self.assertLess(tf.math.reduce_min(probs[1, :]), tf.math.reduce_min(warped_prob_smooth[1, :]))
 
-    def _get_repetition_penalty_inputs(self):
+    @parameterized.expand([(False,), (True,)])
+    def test_repetition_penalty_dist_process(self, use_xla):
```
Comment (gante) [inline tests/generation/test_generation_tf_logits_process.py:110]: changed this one from a two-test format (added in the previous PR) to the parametrized format

File: tests/generation/test_generation_tf_logits_process.py [hunk 2]
```diff
@@ -127,29 +136,21 @@ def _check_repetition_penalty_outputs(self, scores, vocab_size):
...
+    @parameterized.expand([(False,), (True,)])
+    def test_top_k_dist_warper(self, use_xla):
         input_ids = None
         vocab_size = 10
         batch_size = 2
 
         # create ramp distribution
-        ramp_logits = np.broadcast_to(np.arange(vocab_size)[None, :], (batch_size, vocab_size)).copy()
+        ramp_logits = np.broadcast_to(
+            np.arange(vocab_size, dtype=np.float32)[None, :], (batch_size, vocab_size)
```
Comment (Rocketknight1) [inline tests/generation/test_generation_tf_logits_process.py:147]:
```suggestion
            np.arange(vocab_size, dtype=np.float32), (batch_size, vocab_size)
```
You'll never win at [Sasha's Torch golf](https://twitter.com/srush_nlp/status/1516781757596680194) if you manually add dimensions that broadcasting would handle for you!

(Yes, this is the most pointless nit I've ever suggested in a code review)

File: tests/generation/test_generation_tf_logits_process.py [hunk 3]
```diff
@@ -162,18 +163,21 @@ def test_top_k_dist_warper(self):
...
         top_k_warp_safety_check = TFTopKLogitsWarper(top_k=1, filter_value=0.0, min_tokens_to_keep=3)
+        if use_xla:
+            top_k_warp_safety_check = tf.function(top_k_warp_safety_check, jit_compile=True)
 
         scores = top_k_warp_safety_check(input_ids, logits)
         # uniform dist is not changed
         self.assertListEqual(tf.math.reduce_sum(tf.where(scores == 0.0, 1, 0), axis=-1).numpy().tolist(), [0, 0])
 
-        ramp_logits = np.broadcast_to(np.arange(length)[None, :], (batch_size, length)).copy()
+        ramp_logits = np.broadcast_to(np.arange(length, dtype=np.float32)[None, :], (batch_size, length)).copy()
```
Comment (Rocketknight1) [inline tests/generation/test_generation_tf_logits_process.py:173]:
```suggestion
        ramp_logits = np.broadcast_to(np.arange(length, dtype=np.float32), (batch_size, length)).copy()
```

**Review by Rocketknight1 — APPROVED** [review]
Looks good to me!

(Also you don't actually have to add my suggestions, I'm just being a jerk)

**Review by patrickvonplaten — APPROVED** [review]

File: tests/generation/test_generation_tf_logits_process.py (hunk 2 above)
Comment (gante) [reply tests/generation/test_generation_tf_logits_process.py:147]: 😂 Masters of tensor manipulation
